src/Playwright/Core/Waiter.cs

```diff
@@ -243,7 +267,11 @@ private static async Task WrapActionAsync(Func<Task> action, CancellationTokenSo
         }
         catch
         {
-            cts.Cancel();
+            if (!cts.IsCancellationRequested)
```

Comment:
Why this change? What breaks if we always cancel here?

Reply:
There is no need to cancel 'harder'. I have replicated the solution from the other place in this PR where this check is made, and I remembered that I have added this cancellation here some time ago. This is the correct solution for the overall problem. Any code should be written in the most proper/correct way. Simply working is not ever good enough.